src/components/Mail.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { inputEmailActionCreator } from '../redux/actionCreators';
+import Input from './InputClass';
+
+function Mail(props) {
+    function handleChange(evt) {
+        let newText = evt.target.value;
+        props.dispatch(inputEmailActionCreator(newText))
+    }
+    let input = new Input()
+    return (input.render(props, handleChange))
```

Comment:
it is not a React style, please replace 
` return (input.render(props, handleChange))`
with something like
`return (
<Input 
    onChange=handleChange...
/>
);`